### ISSUE TITLE
exportRecordsTyped fails when text field contains unicode non-breaking space

### ISSUE THREAD

**Issue by jwilliman** [vubiostat/redcapAPI#297]
Hi

I'm not sure if this can be considered an issue with with redcapAPI, but I thought I'd flag it as it took me a while to troubleshoot.

If the any of the fields being retrieved contains a non-breaking space (ASCII code 160) `redcapAPI::exportRecordsTyped` fails with the error message 

    Error in if (trimws(as.character(response)) == "") { : 
      missing value where TRUE/FALSE needed

If the record is downloaded as a csv and opened in a text editor (not MS excel) these fields appear to end with the characters '\xa0'. In REDCap the fields appear completely empty. I'm not exactly sure how this data was entered into the project I'm working on, perhaps by copying and pasting from a pdf?

This appears to be because the `httr::POST` command responds with a status of 200 but a content of `NA`

```{r)

> httr::POST(url = rcon$url, body = c(list(token = rcon$token), 
+                                     body), config = c(rcon$config, config))
Response [https://redcap.otago.ac.nz/api/]
  Date: 2024-01-03 01:30
  Status: 200
  Content-Type: text/csv; charset=utf-8
  Size: 62 B
NA

```

That's as far as I can troubleshoot! 

Perhaps just a helpful error message to cover this situation?

Thanks :)

**Comment by spgarbet** [comment]
I think this is related (or maybe the same) as #290 . Encoding issues when reading csv data in R is very problematic. REDCap is not stringent on respecting it's reported encoding. We're pursuing some possibilities to get this working in the API.

**Comment by spgarbet** [comment]
@jwilliman Can you install 2.8.4 `devtools::install_github("vubiostat/redcapAPI")` and see if this fixes your issue?

**Comment by jwilliman** [comment]
I'll have a go. I removed all the unicode characters from the original database and am not sure how they were originally entered in the first place so it make take a while before I can reproduce the error!

**Comment by spgarbet** [comment]
@jwilliman I think I've found how it happened, at least one plausible explanation. If at one point the HTTP header from the server did not contain the proper encoding it defaults to latin1, which then an incorrect transcode would occur. The backend assumes utf-8, and stores as such. The upshot is that one ends up with a database with corrupt characters. I tried under the current version of REDCap to reproduce but I was unable, as it prevent me doing such things. It also sets charset properly on all transactions. I suspect it was a bug in a prior version of the REDCap server.
